test/lib/clients/inst_productsources_test.rb

```diff
@@ -110,4 +111,96 @@ def list_of_repos
       client.CreateSource("http://yast.rulezz.com", "/prod1", "main", nil)
     end
   end
+
+  describe "#NormalizeURL" do
+    it "removes all slashes at the end of the url" do
+      expect(client.NormalizeURL("http://test.suse.de/test_dir///")).to eq(
+        "http://test.suse.de/test_dir"
+      )
+    end
+
+    it "unescape URL" do
+      expect(client.NormalizeURL(
+               "http%3a%2f%2fsome.nice.url%2f%3awith%3a%2f%24p#ci%26l%2fch%40rs%2f"
+      ))
+        .to eq("http://some.nice.url/:with:/$p#ci&l/ch@rs")
+    end
+  end
+
+  describe "#IsAddOnAlreadySelected" do
+    let(:add_on_products) do
+      [
+        {
+          "media"       => 0,
+          "media_url"   => "cd:/?devices=/dev/disk/by-id/ata-QEMU_DVD-ROM_QM00003",
+          "product_dir" => "/"
+        },
+        {
+          "media"       => 1,
+          "media_url"   => "http://download.opensuse.org/debug/distribution/leap/15.1/repo/oss/",
+          "product_dir" => ""
+        },
+        {
+          "media"       => 3,
+          "media_url"   => "http://download.opensuse.org/debug/update/leap/15.1/oss/",
+          "product_dir" => ""
+        },
+        {
+          "media"       => 4,
+          "media_url"   => "http://download.opensuse.org/debug/update/leap/15.1/non-oss/",
+          "product_dir" => ""
+        }
+      ]
+    end
+
+    before do
+      allow(Yast::AddOnProduct).to receive(:add_on_products).and_return(add_on_products)
+    end
+
+    context "is already added" do
+      it "returns source id" do
```

Comment:
NP: Um, what the the test does is actually checking that the empty or default product path works correctly. So maybe something like this would be better:

```suggestion
      it "handles the default and empty product directory correctly" do
```